batch/testResultsCollector.js

```diff
@@ -1,40 +1,52 @@
 import {XUnitParser} from './parsers/xUnitParser';
-import mongoose from 'mongoose';
-import {TestSuiteModel} from '../models/testSuiteModel';
-import {TestCaseModel} from '../models/testCaseModel';
-
+import {GithubHelper} from '../helpers/githubHelper';
+import LudwigDAO from '../database/ludwigDAO';
 
 class TestResultsCollector {
-	constructor(configuration) {
-		this.configuration = configuration;
-		mongoose.connect(configuration.mongo.uri, configuration.mongo.options);
+	constructor(ludwigConfiguration) {
+		this.configuration = ludwigConfiguration;
+		this._xUnitParser = new XUnitParser(this.configuration);
+		this._githubHelper = new GithubHelper(this.configuration);
+		this._dao = LudwigDAO;
+	}
+
+	get parser() {
+		return this._xUnitParser;
+	}
+
+	get githubHelper() {
+		return this._githubHelper;
+	}
+
+	get dao() {
+		return this._dao;
 	}
 
 	saveFromXUnitData(xUnitFilePath, callback) {
-		const parser = new XUnitParser(this.configuration);
-		parser.parse(xUnitFilePath, (errors, parsedTestSuiteData) => {
-			let testSuite = new TestSuiteModel({
-				name: parsedTestSuiteData.name,
-				failures: parsedTestSuiteData.failures,
-				timestamp: parsedTestSuiteData.timestamp
+		const githubHelper = this.githubHelper;
+		this.parser.parseTestSuiteFromFile(xUnitFilePath).then((parsedTestSuiteData) => {
+			const testCasePromises = [];
+			parsedTestSuiteData.testCases.forEach((testCase) => {
+				testCasePromises.push(githubHelper.getFirstCommitForFile(testCase.location));
 			});
 
-			testSuite.save((err) => {
-				if (!err) {
-					TestCaseModel.collection.insert(parsedTestSuiteData.testCases, (err, testCases) => {
-						if (!err) {
-							testSuite.testCases = testCases.ops;
-							testSuite.save((err, data) => {
-								callback(err, data);
-							});
-						} else {
-							callback(err);
-						}
+			Promise.all(testCasePromises)
+				.then((values) => {
+					parsedTestSuiteData.testCases.forEach((value, index) => {
+						value.author = values[index].commit.author;
+						value.author.githubId = values[index].author.id;
 					});
-				} else {
+
+					return this.dao.saveCompleteTestSuite(parsedTestSuiteData);
+				})
+				.then((savedData) => {
+					callback(null, savedData);
```

Comment:
Why keep the `callback` signature while you use Promises inside? Please keep async handling consistent.